packages/cli/templates/react/src/components/FieldPluginExample/index.spec.tsx

```diff
@@ -1,43 +1,166 @@
+import {
+  isAssetModalChangeMessage,
+  isGetContextMessage,
+  isHeightChangeMessage,
+  isModalChangeMessage,
+  isPluginLoadedMessage,
+  isValueChangeMessage,
+} from '@storyblok/field-plugin'
 import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
 import { describe, test, expect, vi } from 'vitest'
 
 import FieldPlugin from '.'
-import { useFieldPlugin } from '@storyblok/field-plugin/react'
-import { FieldPluginResponse } from '@storyblok/field-plugin'
-
-vi.mock('@storyblok/field-plugin/react')
-
-const fieldPluginDefault: FieldPluginResponse<number | undefined> = {
-  type: 'loaded',
-  data: {
-    isModalOpen: false,
-    content: undefined,
-    options: {},
-    spaceId: undefined,
-    storyLang: '',
+
+const fakeContainer = (listener) => {
+  const schema = {
+    field_type: 'test-field-plugin',
+    options: [],
+  }
+  let height = undefined
+  let uid = 'test-uid'
+  let model = undefined
+  let isModalOpen = false
+  let blockId = 'test-block-id'
+  let language = 'default'
+  let storyId = 'test-story-id'
+  let spaceId = 'test-space-id'
+  let token = 'test-token'
+
+  const stateMessage = ({ action, callbackId }) => ({
+    callbackId: callbackId,
+    schema,
+    model,
+    isModalOpen,
+    uid,
+    blockId,
     story: {
       content: {},
     },
-    storyId: undefined,
-    blockUid: undefined,
-    token: undefined,
-    uid: '',
-  },
-  actions: {
-    setContent: vi.fn(),
-    setModalOpen: vi.fn(),
-    requestContext: vi.fn(),
-    selectAsset: vi.fn(),
-  },
+    language,
+    storyId,
+    spaceId,
+    token,
+    action,
+  })
+
+  return {
+    got: ({ data, origin }) => {
+      if (isPluginLoadedMessage(data)) {
+        listener(
+          stateMessage({
+            action: 'loaded',
+            callbackId: data.callbackId,
+          }),
```

Comment:
instead of if --> if else --> if else ..... I think early exit of the function could make this logic a bit more readable. 

```
      
      if (isPluginLoadedMessage(data)) {
        listener(
          stateMessage({
            action: 'loaded',
            callbackId: data.callbackId,
          }),
           return;
          }
          
      if(isValueChangeMessage(data)){
            ....
            return;
          }
```